model/riscv_insts_vext_arith.sail

```diff
@@ -0,0 +1,531 @@
+
+function clause execute(VMVRTYPE(vs2, simm, vd)) = {
+  let SEW     : int = int_power(2, get_sew_pow());
+  let imm_val : int = unsigned(EXTZ(sizeof(xlen), simm));
+  let EMUL    : int = imm_val + 1;
+
+  assert(8 <= SEW & SEW <= 64);
+  assert(EMUL == 1 | EMUL == 2 | EMUL == 4 | EMUL == 8);
+  let EMUL_pow : int = log2(EMUL);
+  let num_elem : int = get_num_elem(EMUL_pow, SEW);
+  let 'n = num_elem;
+  let 'm = SEW;
+
+  let vs2_val : vector('n, dec, bits('m)) = read_vreg(num_elem, SEW, EMUL_pow, vs2);
+  write_vreg(num_elem, SEW, EMUL_pow, vd, vs2_val);
+  vstart = EXTZ(0b0);
+
+  RETIRE_SUCCESS
+} 
+
+mapping simm_string : bits(5) <-> string = {
+  0b00000 <-> "1",
+  0b00001 <-> "2",
+  0b00011 <-> "4",
+  0b00111 <-> "8"
+}
+
+mapping clause assembly = VMVRTYPE(vs2, simm, vd)
+  <-> "vmv" ^ simm_string(simm) ^ "r.v" ^ spc() ^ vreg_name(vd) ^ sep() ^ vreg_name(vs2)
+
+
+/* *******************************OPMVV(VWXUNARY0)******************************** */
+
+union clause ast = VMVXS : (regidx, regidx)
+
+mapping clause encdec = VMVXS(vs2, rd)
+  <-> 0b010000 @ 0b1 @ vs2 @ 0b00000 @ 0b010 @ rd @ 0b1010111
+
+function clause execute(VMVXS(vs2, rd)) = {
+  let SEW      : int = int_power(2, get_sew_pow());
+  let num_elem : int = get_num_elem(0, SEW);
+
+  assert(8 <= SEW & SEW <= 64);
+  assert(num_elem > 0);
+  let 'n = num_elem;
+  let 'm = SEW;
+
+  let vs2_val : vector('n, dec, bits('m)) = read_vreg(num_elem, SEW, 0, vs2);
+  if sizeof(xlen) < SEW then X(rd) = slice(vs2_val[0], 0, sizeof(xlen))
+  else if sizeof(xlen) > SEW then X(rd) = EXTS(vs2_val[0])
+  else X(rd) = vs2_val[0];
+  vstart = EXTZ(0b0);
+
+  RETIRE_SUCCESS
+}
+
+mapping clause assembly = VMVXS(vs2, rd)
+  <-> "vmv.x.s" ^ spc() ^ reg_name(rd) ^ sep() ^ vreg_name(vs2)
+
+
+/* *******************************OPMVX(VRXUNARY0)******************************** */
+
+union clause ast = VMVSX : (regidx, regidx)
+
+mapping clause encdec = VMVSX(rs1, vd)
+  <-> 0b010000 @ 0b1 @ 0b00000 @ rs1 @ 0b110 @ vd @ 0b1010111
+
+function clause execute(VMVSX(rs1, vd)) = {
+  let SEW      : int = int_power(2, get_sew_pow());
+  let num_elem : int = get_num_elem(0, SEW);
+
+  assert(8 <= SEW & SEW <= 64);
+  assert(num_elem > 0);
+  let 'n = num_elem;
+  let 'm = SEW;
+
+  let vm_val  : vector('n, dec, bool)     = read_vmask(num_elem, 0b1, vreg_name("v0"));
+  let rs1_val = get_scalar(rs1, 'm);
+  let vd_val  : vector('n, dec, bits('m)) = read_vreg(num_elem, SEW, 0, vd);
+  result      : vector('n, dec, bits('m)) = undefined;
+  mask_helper : vector('n, dec, bool)     = undefined;
+
+  (result, mask_helper) = init_masked_result(num_elem, SEW, 0, vd_val, vm_val);
+
+  /* one body element */
+  if mask_helper[0] == true then result[0] = rs1_val;
+
+  /* others treated as tail elements */
+  let tail_ag : agtype = get_vtype_vta();
+  if tail_ag == UNDISTURBED then {
+    foreach (i from 1 to (num_elem - 1)) result[i] = vd_val[i]
+  }
+  else if tail_ag == AGNOSTIC then {
```

Comment:
Don't split across lines

Also both bodies are the same? Is that intended?

Reply:
`When a set is marked agnostic, the corresponding set of destination elements in any vector destination operand can either retain the value they previously held, or are overwritten with 1s.` The current tail agnostic policy is to retain the previous value which is the same as the tail undisturbed policy, so both bodies are the same. It can also be modified as to be overwritten with 1s if needed.

Reply:
Well, that's going to become a configuration thing, so either implement both behaviours or don't duplicate code. Having two copies of the same thing, especially with no comment as to why they're the same, is confusing and looks like a mistake.

Reply:
Comments are added to the code involving "agnostic" currently and both behaviors will be implemented when Sail model configuration is enabled.